mutation_writer/multishard_writer.cc

```diff
@@ -181,14 +192,18 @@ future<uint64_t> multishard_writer::operator()() {
         return wait_pending_consumers();
     }).then([this] {
         return _consumed_partitions;
+    }).handle_exception([this] (std::exception_ptr ep) {
+        _on_fail(ep);
```

Comment:
I do not understand why you need a callback if you propagate the error to the caller by returning it. Juts log it at the caller.

Reply:
Because the future returned from the writer is accessed only in wait_for_writer_done when we call stop().

I think I can find a way to avoid the callback. 

I will send a new version.